airbyte-webapp/src/views/Connection/CatalogTree/CatalogTree.tsx

```diff
@@ -1,59 +1,65 @@
 interface CatalogTreeProps {
   streams: SyncSchemaStream[];
-  onChangeStream: (stream: SyncSchemaStream) => void;
+  onStreamsChanged: (streams: SyncSchemaStream[]) => void;
+  isLoading: boolean;
 }
 
-const CatalogTree: React.FC<CatalogTreeProps> = ({ streams, onChangeStream }) => {
+const CatalogTreeComponent: React.FC<React.PropsWithChildren<CatalogTreeProps>> = ({
+  streams,
+  onStreamsChanged,
+  isLoading,
+}) => {
   const { mode } = useConnectionFormService();
-  const onUpdateStream = useCallback(
-    (id: string | undefined, newConfig: Partial<AirbyteStreamConfiguration>) => {
-      const streamNode = streams.find((streamNode) => streamNode.id === id);
 
-      if (streamNode) {
-        const newStreamNode = setIn(streamNode, "config", { ...streamNode.config, ...newConfig });
+  const [searchString, setSearchString] = useState("");
+
+  const onSingleStreamChanged = useCallback(
+    (newValue: SyncSchemaStream) => onStreamsChanged(streams.map((str) => (str.id === newValue.id ? newValue : str))),
+    [streams, onStreamsChanged]
+  );
 
-        onChangeStream(newStreamNode);
-      }
-    },
-    [streams, onChangeStream]
+  const sortedSchema = useMemo(
+    () => streams.sort(naturalComparatorBy((syncStream) => syncStream.stream?.name ?? "")),
+    [streams]
   );
 
-  const { initialValues } = useFormikContext<ConnectionFormValues>();
+  const filteredStreams = useMemo(() => {
+    const filters: Array<(s: SyncSchemaStream) => boolean> = [
+      (_: SyncSchemaStream) => true,
+      searchString
+        ? (stream: SyncSchemaStream) => stream.stream?.name.toLowerCase().includes(searchString.toLowerCase())
+        : null,
+    ].filter(Boolean) as Array<(s: SyncSchemaStream) => boolean>;
 
-  const changedStreams = streams.filter((stream, idx) => {
-    return stream.config?.selected !== initialValues.syncCatalog.streams[idx].config?.selected;
-  });
+    return sortedSchema.filter((stream) => filters.every((f) => f(stream)));
+  }, [searchString, sortedSchema]);
 
   return (
-    <>
-      {streams.map((streamNode) => (
-        <Field key={`schema.streams[${streamNode.id}].config`} name={`schema.streams[${streamNode.id}].config`}>
-          {({ form }: FieldProps<FormikConnectionFormValues>) => (
-            <CatalogSection
-              key={`schema.streams[${streamNode.id}].config`}
-              errors={form.errors}
-              namespaceDefinition={form.values.namespaceDefinition}
-              namespaceFormat={form.values.namespaceFormat}
-              prefix={form.values.prefix}
-              streamNode={streamNode}
-              updateStream={onUpdateStream}
-              changedSelected={changedStreams.includes(streamNode) && mode === "edit"}
-            />
-          )}
-        </Field>
-      ))}
-    </>
+    <BatchEditProvider nodes={streams} update={onStreamsChanged}>
+      <LoadingBackdrop loading={isLoading}>
+        {mode !== "readonly" && <CatalogTreeSearch onSearch={setSearchString} />}
```

Comment:
why shouldn't users search the catalog for a disabled connection?

Reply:
I'm not sure why this was wrapped under readonly in the first place. Maybe it's not needed?

Reply:
It's entirely possible I was a bit too overzealous in shutting down every `input` element when I implemented the "readonly" mode for deleted connections.